ETH-Merge-Bots/Milestones-in-TTD/src/agent.spec.ts

```diff
@@ -0,0 +1,257 @@
+    latestTotalDifficulty,
+    remainingDifficulty: TERMINAL_TOTAL_DIFFICULTY.minus(latestTotalDifficulty).toString(10),
+  };
+};
+
+describe("Milestones in Terminal Total Difficulty Bot", () => {
+  let handleBlock: HandleBlock;
+  let blockEvent: BlockEvent;
+  let avgBlockTime: number;
+
+  beforeEach(async () => {
+    handleBlock = provideHandleBlock(TERMINAL_TOTAL_DIFFICULTY, ETH_BLOCK_DATA, blockCounter);
+    blockEvent = new TestBlockEvent().setTimestamp(0);
+    blockEvent.block.difficulty = new BigNumber("10").toString();
+    avgBlockTime = getAverageBlockTime(ETH_BLOCK_DATA, blockEvent.block.timestamp, 0, 1);
+  });
+
+  it("it should not emit findings when there are more than 20 days until merge", async () => {
+    blockEvent.block.totalDifficulty = new BigNumber("8200000").toString();
+
+    const findings = await handleBlock(blockEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should emit 1 finding when there are less than 20 days and more than 15 days until merge", async () => {
+    blockEvent.block.totalDifficulty = new BigNumber("8300000").toString();
+
+    let findings = await handleBlock(blockEvent);
+
+    const estimatedNumberOfBlocksUntilMerge = getEstimatedNumberOfBlocksUntilMerge(
+      TERMINAL_TOTAL_DIFFICULTY,
+      new BigNumber(blockEvent.block.totalDifficulty),
+      new BigNumber(blockEvent.block.difficulty)
+    );
+
+    const { diffInDays: estimatedNumberOfDaysUntilMerge, estimatedMergeDate } = getNumberOfDays(
+      avgBlockTime,
+      estimatedNumberOfBlocksUntilMerge
+    );
+
+    const mergeInfo = createMergeInfo(
+      estimatedNumberOfDaysUntilMerge,
+      estimatedMergeDate,
+      blockEvent.block.totalDifficulty
+    );
+
+    expect(estimatedNumberOfDaysUntilMerge).toStrictEqual(20);
+    expect(findings).toStrictEqual([createFinding(mergeInfo)]);
+
+    findings = await handleBlock(blockEvent);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should emit 1 finding when there are less than 15 days and more than 10 days until merge", async () => {
+    blockEvent.block.totalDifficulty = new BigNumber("8750000").toString();
+
+    let findings = await handleBlock(blockEvent);
+
+    const estimatedNumberOfBlocksUntilMerge = getEstimatedNumberOfBlocksUntilMerge(
+      TERMINAL_TOTAL_DIFFICULTY,
+      new BigNumber(blockEvent.block.totalDifficulty),
+      new BigNumber(blockEvent.block.difficulty)
+    );
+
+    const { diffInDays: estimatedNumberOfDaysUntilMerge, estimatedMergeDate } = getNumberOfDays(
+      avgBlockTime,
+      estimatedNumberOfBlocksUntilMerge
+    );
+
+    const mergeInfo = createMergeInfo(
+      estimatedNumberOfDaysUntilMerge,
+      estimatedMergeDate,
+      blockEvent.block.totalDifficulty
+    );
+
+    expect(estimatedNumberOfDaysUntilMerge).toStrictEqual(15);
+    expect(findings).toStrictEqual([createFinding(mergeInfo)]);
+
+    findings = await handleBlock(blockEvent);
+    expect(findings).toStrictEqual([]);

```

Comment:
In these lines of the tests, you can increase the `totalDifficulty` a bit to prove that it still doesn't emit a finding, so that it's more realistic, wdyt?

Reply:
sure. added them 👍